### PR TITLE
Add the ability to "Claim" a resource.

### LINKED ISSUE #3831: Add a button to allow a logged in user to claim an item to work on.
The button should be in the show view.

### LINKED ISSUE #3832: Add a facet to show all items claimed by "me" and Unclaimed items.

### PR DESCRIPTION
**Pull request by tpendragon**
Closes #3831 
Closes #3832 

Screen Shot 2020-04-28 at 4 04 44 PM
Screen Shot 2020-04-28 at 4 05 31 PM
Screen Shot 2020-04-28 at 4 07 08 PM
Screen Shot 2020-04-28 at 4 05 40 PM

### CODE REVIEW

**Comment by kelea99** [comment]
Looks fantastic!

**Comment by hackmastera** [comment]
"Claim from Raphael" is a button that allows you to claim an object that's been claimed by another person?

**Comment by kelea99** [comment]
@hackmastera that is what I gathered from this option. It can be useful if folks would like to take something due to language or subject knowledge, or if something was claimed by an individual no longer working in the project, etc..

File: spec/views/catalog/_show_header_default.html.erb_spec.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe "catalog/_show_header_default.html.erb" do
+  context "when the user can't update the resource" do
+    before do
+      allow(view).to receive(:render_document_heading)
+    end
+    it "doesn't show" do
```
Comment (hackmastera) [inline spec/views/catalog/_show_header_default.html.erb_spec.rb:10]: "doesn't show a claim button"?

**Comment by tpendragon** [comment]
@hackmastera Yeah, I'm trying to figure out the test.

I put it on the top because A) The bottom bar felt mostly navigational, B) The bottom bar's full for most resource types, and C) There's a bunch of partials that make up the bottom bar, this was easier.

**Comment by tpendragon** [comment]
@hackmastera Should be ready for re-review.

**Comment by hackmastera** [comment]
> I put it on the top because A) The bottom bar felt mostly navigational, B) The bottom bar's full for most resource types, and C) There's a bunch of partials that make up the bottom bar, this was easier.

Yeah. I'm concerned we really need some more UI attention here. But obviously the need for the feature is urgent so this PR shouldn't be blocked.

**Pull request merged**